packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/api/SelectionDialogDescriptionConverter.java

```diff
@@ -0,0 +1,341 @@
+            var optionalObject = this.objectService.getObject(optionalEditingContext.get(), optionalId.get());
+            if (optionalObject.isPresent()) {
+                result = optionalObject.get();
+            } else {
+                var optionalEditingDomain = Optional.of(optionalEditingContext.get())
+                        .filter(IEMFEditingContext.class::isInstance)
+                        .map(IEMFEditingContext.class::cast)
+                        .map(IEMFEditingContext::getDomain);
+
+                if (optionalEditingDomain.isPresent()) {
+                    var editingDomain = optionalEditingDomain.get();
+                    ResourceSet resourceSet = editingDomain.getResourceSet();
+                    URI uri = new JSONResourceFactory().createResourceURI(optionalId.get());
+
+                    result = resourceSet.getResources().stream()
+                            .filter(resource -> resource.getURI().equals(uri)).
+                            findFirst()
+                            .orElse(null);
+                }
+            }
+        }
+
+        return result;
+    }
+
+
+    private Object getParentObject(VariableManager variableManager) {
+        Object result = null;
+        Object self = variableManager.getVariables().get(VariableManager.SELF);
+        if (self instanceof EObject eObject) {
+            Object semanticContainer = eObject.eContainer();
+            if (semanticContainer == null) {
+                semanticContainer = eObject.eResource();
+            }
+            result = semanticContainer;
+        }
+        return result;
+    }
+
+    private List<?> getChildren(VariableManager variableManager, AQLInterpreter interpreter, SelectionDialogTreeDescription selectionDialogTreeDescription, Function<VariableManager, Boolean> isSelectableProvider) {
+        List<Object> result = new ArrayList<>();
+
+        List<String> expandedIds = new ArrayList<>();
+        Object objects = variableManager.getVariables().get(TreeRenderer.EXPANDED);
+        if (objects instanceof List<?> list) {
+            expandedIds = list.stream().filter(String.class::isInstance).map(String.class::cast).toList();
+        }
+
+        String id = this.getTreeItemId(variableManager);
+        if (expandedIds.contains(id)) {
+            result.addAll(this.computeChildrenFromExpression(variableManager, interpreter, selectionDialogTreeDescription));
+        }
+        result.removeIf(object -> {
+            if (object instanceof EObject eObject) {
+                VariableManager childVariableManager = variableManager.createChild();
+                childVariableManager.put(VariableManager.SELF, eObject);
+                return !isSelectableProvider.apply(childVariableManager) && !this.hasChildren(interpreter, selectionDialogTreeDescription, childVariableManager, isSelectableProvider);
+            } else {
+                return false;
+            }
+        });
```

Comment:
You can't decide that, let the specifiers display what they want. And you can't decide that without the MASSIVE performance impact from before.

Reply:
Done